crates/biome_js_analyze/src/analyzers/style/use_numeric_literals.rs

```diff
@@ -72,13 +75,19 @@ impl Rule for UseNumericLiterals {
 
     fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
         let node = ctx.query();
-
+        let an_or_a = if matches!(state.radix, Radix::Octal) {
+            "an"
+        } else {
+            "a"
+        };
```

Comment:
nit: you could create a `const` method inside `CallInfo`, and the compiler will be able to optimise it. Then we can move the call of that method directly in the markup, like we do with `description()`